### PR TITLE
F r i e n d s

### PR DESCRIPTION
**Pull request by Dalgros**

### CODE REVIEW

File: src/Elan.Data/ElanDbContext.cs [hunk 1]
```diff
@@ -22,6 +24,26 @@ protected override void OnModelCreating(ModelBuilder builder)
...
+
+            friendsRelationBuilder
+                .HasKey(c => new { c.FirstUserId, c.SecondUserId });
+
+            friendsRelationBuilder
+                .HasOne(x => x.FirstUser)
+                .WithMany()
+                .OnDelete(DeleteBehavior.Restrict);
+
+            friendsRelationBuilder
```
[the remark this answers is not in the document]
Comment (Dalgros) [reply src/Elan.Data/ElanDbContext.cs]: done

File: src/Elan.Friends/Contracts/IFriendsService.cs [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+using Elan.Data.Models.Friends;
+using System.Collections.Generic;
 using System.Threading.Tasks;
-using Elan.Data.Models.Account;
 
 namespace Elan.Friends.Contracts
 {
     public interface IFriendsService
     {
+        Task<FriendsRelation> CreateRelation(ElanUser FirstUser, ElanUser SecondUser);
```
[the remark this answers is not in the document]
Comment (Dalgros) [reply src/Elan.Friends/Contracts/IFriendsService.cs]: done

File: src/Elan.Friends/Services/FriendsService.cs [hunk 3]
```diff
@@ -17,14 +19,42 @@ public FriendsService(IDataService dataService)
...
+                SecondUser = secondUser,
+                CreatedDate = DateTime.UtcNow
+            };
+
+            await _dataService.GetSet<FriendsRelation>().AddAsync(relation);
+            await _dataService.SaveDbAsync();
+
+            return relation;
+        }
+
```
[the remark this answers is not in the document]
Comment (Dalgros) [reply src/Elan.Friends/Services/FriendsService.cs]: done

File: src/Elan.Friends/Services/FriendsService.cs [hunk 4]
```diff
@@ -17,14 +19,40 @@ public FriendsService(IDataService dataService)
...
+                .Include(u => u.FirstUser)
+                .Include(u => u.SecondUser)
+                .Where(u => u.FirstUser.Id == user.Id || u.SecondUser.Id == user.Id)
+                .Select(r => GetFriendUser(r, user))
                 .ToListAsync();
-
             return result;
         }
+
+        private ElanUser GetFriendUser(FriendsRelation r, ElanUser user)
```
Comment (gmackiewicz) [inline src/Elan.Friends/Services/FriendsService.cs]: We should use more descriptive names than `r`.

Comment (Dalgros) [reply src/Elan.Friends/Services/FriendsService.cs]: done

**Pull request merged**